crates/rust-analyzer/src/config.rs

```diff
@@ -312,8 +312,10 @@ config_data! {
         /// How to render the size information in a memory layout hover.
         hover_memoryLayout_size: Option<MemoryLayoutHoverRenderKindDef> = Some(MemoryLayoutHoverRenderKindDef::Both),
 
-        /// How many fields of a struct to display when hovering a struct.
-        hover_show_structFields: Option<usize> = None,
+        /// How many variants of an enum to display when hovering on. Show none if empty.
+        hover_show_enumVariants: Option<usize> = Some(5),
+        /// How many fields of a struct or union to display when hovering on. Show none if empty.
+        hover_show_structOrUnionFields: Option<usize> = Some(5),
```

Comment:
```suggestion
        hover_show_fields: Option<usize> = Some(5),
```
since this should also work for enum variants, so just fields should be good enough